capstone/client/src/components/ListPage.js

```diff
@@ -226,10 +229,48 @@ class ListPage extends Component {
     }
   }
 
+  selectList = (event) => {
+    if (this.selectedItems.size !== 0) {
+      for (let val of this.selectedItems) {
+        this.itemsToComponent[val].click();
+      }
+    }
+    this.setState({
+      listId : this.state.userLists[event.target.name].listId,
+    });
+    for (const i in this.state.userLists[event.target.name].itemTypes) {
+      this.itemsToComponent[this.state.userLists[event.target.name].itemTypes[i]].click();
+    }
+  }
+
+  userLists() {
+    if(this.state.userId !== -1 && this.state.userLists == null) {
+      axios.get('/api/v1/get-user-lists', { params : { userId : this.state.userId }})
+        .then(res => {
+          this.setState({
+            totalLists: res.data.userLists.length,
+            userLists: res.data.userLists
+          });
+        });
+    }
+    if(this.userLists !== null) {
```

Comment:
One way of doing this is to save the list items in the state when a user clicks on a list. In the render function, you can modify the assignment of checkboxItems so an item gets mapped to a checked version of the FormControlLabel if that items is contained in the current list. 
(this.state.currentSavedItems ? "checked component" : "non-checked component")